ga4gh/datamodel/__init__.py

```diff
@@ -56,6 +81,12 @@ class PysamDatamodelMixin(object):
 
     maxStringLength = 2**10  # arbitrary
 
+    # LRU cache of open file handles
+    cache = PysamFileHandleCache()
```

Comment:
I tend to agree --- we're not encapsulating the cache concept very well here, and spreading the implementation across the two classes. More detailed comments in a minute when I figure it out.